snafu/benchmarks/_benchmark.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Base benchmark tools."""
+from abc import ABC, abstractmethod
+from typing import Any, Dict, Iterable
+from dataclasses import dataclass
+import logging
+import json
+from snafu import registry
+from snafu.config import Config, ConfigArgument
+
+
+@dataclass
+class BenchmarkResult:
+    name: str
+    metadata: Dict[str, Any]
+    config: Dict[str, Any]
+    data: Dict[str, Any]
+    label: str
+
+    def to_json(self) -> str:
+        result: Dict[str, Any] = dict()
+        result.update(self.config)
```

Comment:
A lot of benchmarks will add their config mappings on the top-level JSON output next to the result data, rather that placing them in a separate ``config`` key, which is why I formatted the JSON result like this. But if we want to change the structure of the benchmark result to something different then here is where we can do that without having to modify each benchmark.